src/environment.jl

```diff
@@ -341,3 +341,28 @@ function has_cuda()
         return parse(Bool, flag)
     end
 end
+
+"""
+    MPI.has_rocm()
+
+Check if the MPI implementation is known to have ROCm support. Currently only Open MPI
+provides a mechanism to check, so it will return `false` with other implementations
+(unless overriden).
+
+This can be overriden by setting the `JULIA_MPI_HAS_ROCM` environment variable to `true`
+or `false`.
+"""
+function has_rocm()
```

Comment:
Maybe we can add a third function `has_gpu()` that calls `has_cuda() || has_rocm()`